compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -1293,9 +1296,20 @@ impl<'a, G: EmissionGuarantee> Diag<'a, G> {
     /// `cancel`, etc. Afterwards, `drop` is the only code that will be run on
     /// `self`.
     fn take_diag(&mut self) -> DiagInner {
+        if let Some(path) = &self.long_ty_path {
+            self.note(format!(
+                "the full name for the type has been written to '{}'",
+                path.display()
+            ));
+            self.note("consider using `--verbose` to print the full type name to the console");
+        }
         Box::into_inner(self.diag.take().unwrap())
     }
```

Comment:
Now there's no chance of forgetting to print this path (although we need to remember to use `long_ty_path()` to set the path right after error construction if we've made short strings before that).

Reply:
Suggestion: could you add that remark as a doc comment for `long_ty_path` immediately below?

Reply:
Added a bunch more to the docs of `long_ty_path` *and?* `short_string` explaining how they are meant to be used in tandem.